cmd/template/nodepool/provider/azure.go

```diff
@@ -73,6 +75,18 @@ func WriteAzureTemplate(out io.Writer, config NodePoolCRsConfig) error {
 }
 
 func newAzureMachinePoolCR(config NodePoolCRsConfig) *expcapzv1alpha3.AzureMachinePool {
+	var spot *v1alpha3.SpotVMOptions
+	if config.AzureUseSpotVms {
+		var maxPrice *resource.Quantity
+		if config.AzureSpotMaxPrice > 0 {
+			maxPriceScalar := resource.MustParse(fmt.Sprintf("%f", config.AzureSpotMaxPrice))
+			maxPrice = &maxPriceScalar
+		}
+		spot = &v1alpha3.SpotVMOptions{
+			MaxPrice: maxPrice,
```

Comment:
Is leaving this as `nil` the same as setting it to `-1` (using on-demand pricing as max)?

Reply:
yeah. When setting `-1` on happa, it becomes a `nil` in the AzureMachinePool CR

Reply:
Not currently
https://github.com/giantswarm/api/blob/2125e0f95ea111fe9c43cdd477d9f68d32959338/service/nodepool/creator/service.go#L661

Reply:
Fuck you are right:

https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachinescalesets?tabs=json#BillingProfile

-1 is the needed value

Reply:
done